docs: add configuration guide

AndreyBykov opened the pull request.

mnmkng commented:
I would put the crawlee.json config first, then env vars, then config in code.

The reason being, configuration usually refers to options you can configure externally, like in CI, on install, on deploy etc so the in-code config is actually the least important, because the point of config is not to have to touch code.

AndreyBykov commented:
Updated. Removed dedicated env vars guide (included everything to the Configuration guide).

In case you wanna see how it looks rendered, attaching the screenshot:

![Screenshot 2022-08-04 at 02-19-08 Configuration Crawlee](https://user-images.githubusercontent.com/18262743/182737060-3aaec87b-b2f0-44c7-9b0a-b5d3ad5fb3bd.png)

B4nan commented on docs/quick-start/index.mdx:
```diff
@@ -124,7 +124,7 @@ You can find more examples showcasing various features of Crawlee in the [Exampl
 
 **Related links**
 
-- [Environment variables](./guides/environment-variables)
```
we should setup a redirect for the old URL

https://github.com/apify/crawlee/blob/master/website/docusaurus.config.js#L101

AndreyBykov replied:
Added - to `/docs/guides/configuration#environment-variables`

AndreyBykov replied:
oh, you can't really redirect to `/docs/guides/configuration#environment-variables` :/ `/docs/guides/configuration` it is 🤷‍♂️

B4nan replied:
you would have to use different approach for that, its possible but not worth it here iguess

AndreyBykov commented on docs/guides/configuration.mdx:
```diff
@@ -0,0 +1,223 @@
+    // and abort the run
+    if (request.url === 'https://www.example.com/2') {
+        await sleep(10_000);
+        process.exit(0);
```
In this example, though, it's intentional that I forcefully end the process...
